### PR TITLE
NuGet update and Dbreeze serialization fix

### PR DESCRIPTION
**Pull request by codingupastorm**
Allows the sidechains repo to serialize everything given the changes to the FN Dbreeze serialization and ultimately allows multiple nodes to run in the one process by removing static dependencies.

I haven't actually connected to the network and completed a transfer yet - if someone could do that it would be ace.

### CODE REVIEW

File: src/Stratis.FederatedPeg.Features.FederationGateway/TargetChain/CrossChainTransferStore.cs [hunk 1]
```diff
@@ -1083,7 +1088,10 @@ private void PutTransfers(DBreeze.Transactions.Transaction dbreezeTransaction, I
 
             // Write each transfer in order.
             foreach (ICrossChainTransfer transfer in orderedTransfers)
-                dbreezeTransaction.Insert(transferTableName, transfer.DepositTransactionId.ToBytes(), transfer);
+            {
+                byte[] transferBytes = this.dBreezeSerializer.Serialize(transfer);
```
Comment (quantumagi) [inline src/Stratis.FederatedPeg.Features.FederationGateway/TargetChain/CrossChainTransferStore.cs]: `this.dbreezeSerializer` is not required if doing it this way:

`byte[] crossChainTransferBytes = crossChainTransfer.ToBytes(this.network.Consensus.ConsensusFactory);`

File: src/Stratis.FederatedPeg.Features.FederationGateway/TargetChain/CrossChainTransferStore.cs [hunk 2]
```diff
@@ -1066,7 +1069,9 @@ private void PutTransfer(DBreeze.Transactions.Transaction dbreezeTransaction, IC
         {
             Guard.NotNull(crossChainTransfer, nameof(crossChainTransfer));
 
-            dbreezeTransaction.Insert<byte[], ICrossChainTransfer>(transferTableName, crossChainTransfer.DepositTransactionId.ToBytes(), crossChainTransfer);
+            byte[] crossChainTransferBytes = this.dBreezeSerializer.Serialize(crossChainTransfer);
```
Comment (quantumagi) [inline src/Stratis.FederatedPeg.Features.FederationGateway/TargetChain/CrossChainTransferStore.cs]: this.dbreezeSerializer is not required if doing it this way:

byte[] crossChainTransferBytes = crossChainTransfer.ToBytes(this.network.Consensus.ConsensusFactory);

File: src/Stratis.FederatedPeg.Features.FederationGateway/TargetChain/CrossChainTransferStore.cs [hunk 3]
```diff
@@ -69,7 +71,7 @@ public class CrossChainTransferStore : ICrossChainTransferStore
 
         public CrossChainTransferStore(Network network, DataFolder dataFolder, ConcurrentChain chain, IFederationGatewaySettings settings, IDateTimeProvider dateTimeProvider,
             ILoggerFactory loggerFactory, IWithdrawalExtractor withdrawalExtractor, IFullNode fullNode, IBlockRepository blockRepository,
-            IFederationWalletManager federationWalletManager, IFederationWalletTransactionHandler federationWalletTransactionHandler)
+            IFederationWalletManager federationWalletManager, IFederationWalletTransactionHandler federationWalletTransactionHandler, DBreezeSerializer dBreezeSerializer)
```
Comment (quantumagi) [inline src/Stratis.FederatedPeg.Features.FederationGateway/TargetChain/CrossChainTransferStore.cs]: dbreezeSerializer should not be required. See below.

File: src/Stratis.FederatedPeg.Tests/CrossChainTestBase.cs [hunk 4]
```diff
@@ -84,6 +82,7 @@ public CrossChainTestBase()
             this.walletFeePolicy = Substitute.For<IWalletFeePolicy>();
             this.nodeLifetime = new NodeLifetime();
             this.connectionManager = Substitute.For<IConnectionManager>();
+            this.dBreezeSerializer = new DBreezeSerializer(this.network);
```
Comment (quantumagi) [inline src/Stratis.FederatedPeg.Tests/CrossChainTestBase.cs]: Changes to tests should not be required. See above.

**Comment by codingupastorm** [comment]
@quantumagi fair point, but I think keeping the serializer in:
- Is slightly more explicit
- Avoids the objects being in charge of serializing themselves
- Avoids the work being redone

**Comment by quantumagi** [comment]
@codingupastorm , ok makes sense. the more control the better. good job.

**Pull request merged**